ipatests/test_integration/test_commands.py

```diff
@@ -1489,3 +1489,26 @@ def test_ipa_cacert_manage_prune(self):
         self.master.run_command(['date', '-s', '-15Years'])
 
         assert isrgrootx1_nick in result
+
+    def test_client_doesnot_throw_responsenotready_error(self):
+        """
+        This testcase checks that ipa command
+        doesn't throw http.client.ResponseNotReady error
+        when current users session is deleted from the cache
+        """
+        error_msg = (
+            "http.client.ResponseNotReady: Request-sent \n"
+            "ipa: ERROR: an internal error has occurred"
+        )
+
+        # kinit as admin on ipa-client and run ipa command
+        tasks.kinit_admin(self.clients[0], raiseonerr=False)
+        self.clients[0].run_command(['ipa', 'user-find'])
+
+        # Delete the current user session cache on IPA server
+        self.master.run_command(['rm', '-fv', '/run/ipa/ccaches/*'])
```

Comment:
According to https://github.com/encukou/pytest-multihost/blob/master/pytest_multihost/host.py#L262, run_command with a list of args does not provide shell expansion. You can either replace your line with `run_command("rm -fv /run/ipa/ccaches/*")` or do as in https://github.com/freeipa/freeipa/blob/master/ipatests/pytest_ipa/integration/tasks.py#L938